pwndbg/commands/leakfind.py

```diff
@@ -0,0 +1,155 @@
+
+#Used to recursively print the pointer chain. 
+#addr is a pointer. It is taken to be a child pointer.
+#visitedMap is a map of children -> (parent,parent_start)
+def get_rec_addr_string(addr,visitedMap):
+    page = getPage(addr)
+    if not page is None:
+        if not addr in visitedMap:
+            return ""
+        
+        parentInfo = visitedMap[addr]
+        parent = parentInfo[0]
+        parent_base_addr = parentInfo[1]
+        if parent-parent_base_addr < 0:
+            curText = hex(parent_base_addr) + hex(parent-parent_base_addr)
+        else:
+            curText = hex(parent_base_addr) + "+"+ hex(parent-parent_base_addr)
+        if parent_base_addr == addr:
+            return ""
+        return get_rec_addr_string(parent_base_addr,visitedMap) + M.get(parent_base_addr,text=curText)+arrow_right
+    else:
+        return ""
+
+#Utility function to get a page from an address.
+def getPage(address):
+    return pwndbg.vmmap.find(address)
+
+#Useful for debugging. Prints a map of child -> (parent, parent_start)
+def dbg_print_map(maps):
+    for child, parentInfo in maps.items():
+        print(hex(child) + "("+hex(parentInfo[0])+","+hex(parentInfo[1])+")")
+
+parser = argparse.ArgumentParser()
+parser.description = """
+Attempt to find a leak chain given a starting address. Scans memory near the given address, looks for pointers, 
+and continues that process to attempt to find leaks.\n
+Example: leakfind $rsp filename 0x48 6. This would look for any chains of leaks that point to a section in filename which begin near $rsp, are never 0x48 bytes further from a known pointer, 
+and are a maximum length of 6.\n
+"""
+parser.add_argument("address",help="Starting address to find a leak chain from.")
+parser.add_argument("-p","--page_name",type=str,nargs="?",default=None,help="Substring required to be part of the name of any found pages")
+parser.add_argument("-o","--max_offset",default=0x48,nargs="?",help="Max offset to add to addresses when looking for leak.")
+parser.add_argument("-d","--max_depth",default=0x4,nargs="?",help="Maximum depth to follow pointers to.")
+parser.add_argument("-s","--stride",nargs="?",default=0x1,help="Stride to add to add between pointers considered. For example, if this is 4 it would only consider pointers at an offset divisible by 4 from the starting pointer.")
+parser.add_argument('--negative_offset',nargs="?",default=0x0,help="Max negative offset to search before an address when looking for a leak.")
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+def leakfind(address=-1,page_name=None,max_offset=0x40,max_depth=0x4,stride=0x1,negative_offset=0x0):
+    if address == -1:
+        print("No starting address provided. Please run leakfind -h for more information.")
+        return
+
+    foundPages = getPage(address)
+
+    if not foundPages:
+        print("Starting address is not mapped. Please run leakfind -h for more information.")
```

Comment:
Can you check if `raise argparse.ArgumentTypeError('Starting address is not mapped.')` would be better here?

Same for the other checks.

Reply:
I changed it to do this. Why is this behavior preferable? 
It seems more natural to tell users to run the help function than it does show information about an exception.

Reply:
Hmm I thought it is handled in a way it doesn't show an exception... hmm

Reply:
It's not that the stack trace is directly shown, but it suggests either seeing the stacktrace or using a debugger rather than just reading the help.
Output of `leakfind 0`:
`Exception occured: leakfind: Starting address is not mapped. (<class 'argparse.ArgumentTypeError'>)`
`For more info invoke set exception-verbose on and rerun the command
or debug it by yourself with set exception-debugger on
`

Reply:
We should handle this exception somewhere in ArgparsedCommand then... or fallback to the previous solution.